spark/src/test/java/org/apache/iceberg/spark/actions/TestNewRewriteDataFilesAction.java

```diff
@@ -151,6 +169,52 @@ public void testBinPackPartitionedTable() {
     assertEquals("Rows must match", expectedRecords, actualRecords);
   }
 
+  @Test
```

Comment:
currently just add 1 sanity test, will add more later once we agree upon the interface changes.